ext/standard/math.c

```diff
@@ -90,22 +47,31 @@ static inline double php_intpow10(int power) {
 }
 /* }}} */
 
-/* {{{ php_round_helper
-       Actually performs the rounding of a value to integer in a certain mode */
-static inline double php_round_helper(double value, int mode) {
-	double integral, fractional;
+#define PHP_ROUND_BASIC_EDGE_CASE() do {\
+		if (places > 0) {\
+			edge_case = fabs((integral + copysign(0.5, integral)) / exponent);\
+		} else {\
+			edge_case = fabs((integral + copysign(0.5, integral)) * exponent);\
+		}\
+	} while (0)
+#define PHP_ROUND_ZERO_EDGE_CASE() do {\
+		if (places > 0) {\
+			edge_case = fabs((integral) / exponent);\
+		} else {\
+			edge_case = fabs((integral) * exponent);\
+		}\
+	} while (0)
```

Comment:
I'm not a fan of those sort of macros because they make the code confusing.

`integral` and `exponent` should be arguments to the macro _at minimum_, and I'd prefer if edge case was returned.

Also, why not have these as inline functions (possibly marked with `zend_always_inline`) as this would be IMHO clearer.

Reply:
Thanks, that makes sense. I'll fix it.

Reply:
Fixed those!